Update collaboration-engine documentation to 3.1 for Vaadin 14.7 (v14) (#616)
vaadin/docs: articles/ce/components/collaboration-message-list.asciidoc

---
title: Message List
order: 3
layout: page
---

[[ce.collaboration-message-list]]
= Real-Time Chat and Commenting Components

> You can add a real-time chat to your application by using the
> `CollaborationMessageList` and `CollaborationMessageInput` components.

Reviewer: ```suggestion
You can add a real-time chat to your application by using the `CollaborationMessageList` and `CollaborationMessageInput` components.
```

.Example of a `CollaborationMessageList` and `CollaborationMessageInput`

> image::images/collaboration-messages-example.png[A Chat with three messages, and controls for typing in new messages, width=399]

Reviewer: ```suggestion
image::images/collaboration-messages-example.png["A Chat with three messages, as well as controls for typing in new messages", width=399]
```

> They enhance the regular `MessageList` and `MessageInput` components
> by automatically synchronizing the messages for all users connected to the same topic.
> The message list only renders the submitted messages, while the message input provides
> controls for submitting new messages.
> Here's an example of setting them up:

Reviewer: ```suggestion
They enhance the regular `MessageList` and `MessageInput` components by automatically synchronizing the messages for all users connected to the same topic.
The message list only renders the submitted messages, while the message input provides controls for submitting new messages.

The following example shows how to set them up:
```
